File: ios/PaypalTrigger.m
#import "React/RCTBridgeModule.h"
#import <Foundation/Foundation.h>

@interface RCT_EXTERN_MODULE(PaypalTrigger NSObject)
RCT_EXTERN_METHOD(triggerPayPalCheckout: paymentId (NSString) onApprove:(RCTResponseSenderBlock)onApprove onCancel:(RCTResponseSenderBlock)onCancel onError(RCTResponseSenderBlock)onError)
@end

File: ios/PaypalTrigger.swift
import PayPalCheckout
import Foundation

@objc(PaypalTrigger)
class PaypalTrigger: NSObject {
    @objc 
    func triggerPayPalCheckout(_ id: NSString,  onApprove: @escaping RCTResponseSenderBlock, onCancel: @escaping RCTResponseSenderBlock,  onError: @escaping RCTResponseSenderBlock) {
    Checkout.start(
        createOrder: { createOrderAction in
          createOrderAction.set(orderId: id as String)
        }, onApprove: { approval in
          onApprove([])
        }, onCancel: {
              onCancel([])
        }, onError:  { error in
             onError([error])
        }
    )
  }
}